got krb5 error when cross compile using release v19.1.1 or v19.1.2

**Describe the problem**

we got krb5 errors when using `build/builder.sh mkrelease arm64-linux-gnueabi` for arm64, and this can be reproduced in release v19.1.1 and v19.1.2, maybe all v19.x.x versions. 

error log: 
```
checking for ns_initparse... yes
checking for ns_name_uncompress... yes
checking for dn_skipname... yes
checking for res_search... yes
checking whether pragma weak references are supported... yes
checking for constructor/destructor attribute support... configure: error: Cannot test for constructor/destructor support when cross compiling
Makefile:556: recipe for target '/go/native/aarch64-unknown-linux-gnueabi/krb5/Makefile' failed
make: *** [/go/native/aarch64-unknown-linux-gnueabi/krb5/Makefile] Error 1
```
```
configure: WARNING: Could not find Tcl which is needed for some tests
checking whether ln -s works... yes
checking for aarch64-unknown-linux-gnueabi-ranlib... aarch64-unknown-linux-gnueabi-ranlib
checking for ar... ar cqv
checking for ar... ar cruv
checking for a BSD-compatible install... /usr/bin/install -c
checking for ar... ar
checking for perl... perl
checking for working regcomp... configure: error: Cannot test regcomp when cross compiling
Makefile:556: recipe for target '/go/native/aarch64-unknown-linux-gnueabi/krb5/Makefile' failed
make: *** [/go/native/aarch64-unknown-linux-gnueabi/krb5/Makefile] Error 1
```
```
checking for u_int16_t... yes
checking for u_int32_t... yes
checking for int8_t... yes
checking for int16_t... yes
checking for int32_t... yes
checking for sh... /bin/sh
checking for sh5... false
checking for bash... /bin/bash
checking if /bin/sh supports functions... yes
checking for POSIX printf positional specification support... configure: error: Cannot test for printf positional argument support when cross compiling
Makefile:556: recipe for target '/go/native/aarch64-unknown-linux-gnueabi/krb5/Makefile' failed
make: *** [/go/native/aarch64-unknown-linux-gnueabi/krb5/Makefile] Error 1
```

**To Reproduce**

just run `build/builder.sh mkrelease arm64-linux-gnueabi`, nothing more

**Environment:**
 - CockroachDB version: v19.1.1/v19.1.2
 - Server OS: macos 10.14.5 or ubuntu 18.04, go 1.11

--- a/build/builder/mkrelease.sh
+++ b/build/builder/mkrelease.sh
@@ -39,8 +39,8 @@ case "${1-}" in
       XGOARCH=amd64
       XCMAKE_SYSTEM_NAME=Linux
       TARGET_TRIPLE=x86_64-unknown-linux-gnu
-# -lrt is needed as clock_gettime isn't part of glibc prior to 2.17
-# once we update - the -lrt can be removed
+      # -lrt is needed as clock_gettime isn't part of glibc prior to 2.17.
+      # If we update to a newer glibc, the -lrt can be removed.
       LDFLAGS="-static-libgcc -static-libstdc++ -lrt"
       SUFFIX=-linux-2.6.32-gnu-amd64
     ) ;;
@@ -56,6 +56,11 @@ case "${1-}" in
     ) ;;
 
   ?(arm64-)linux?(-gnueabi))
+    # Manually set the correct values for configure checks that libkrb5 won't be
+    # able to perform because we're cross-compiling.
+    export krb5_cv_attr_constructor_destructor=yes
+    export ac_cv_func_regcomp=yes
+    export ac_cv_printf_positional=yes
     args=(
       XGOOS=linux
       XGOARCH=arm64

